### PR TITLE
[FEAT] Oil Reserves

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description

This PR add the ability: 

- Place Oil Reserve
- Move an Oil Reserve
- Drill an Oil Reserve
- Buy an Oil Drill

An oil reserve will produce 10 oil regularly and then 20% more on every 3rd drill.

oil

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Give yourself a reserve
- Place it
- Buy some drills
- Drill oil

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/events/landExpansion/drillOilReserve.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+};
+
+export const BASE_OIL_DROP_AMOUNT = 10;
+export const OIL_RESERVE_RECOVERY_TIME = 20 * 60 * 60 * 1000;
+
+export function canDrillOilReserve(
+  reserve: OilReserve,
+  now: number = Date.now()
+) {
+  return now - reserve.oil.drilledAt > OIL_RESERVE_RECOVERY_TIME * 1000;
```
Comment (Rofrtd) [inline src/features/game/events/landExpansion/drillOilReserve.ts:23]: Looks like you're multiplying to 1000 twice, 
OIL_RESERVE_RECOVERY_TIME is already Milliseconds.

File: src/features/game/expansion/components/resources/oilReserve/OilReserve.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+export const OilReserve: React.FC<Props> = ({ id }) => {
+  const { gameService } = useContext(Context);
+  const [drilling, setDrilling] = useState(false);
+  const [oiledAmount, setOilAmount] = useState<number>();
+
+  const reserve = useSelector(gameService, _reserve(id), compareResource);
+  const drills = useSelector(gameService, _drills);
+  const timeLeft = getTimeLeft(
```
Comment (Rofrtd) [inline src/features/game/expansion/components/resources/oilReserve/OilReserve.tsx:37]: On this function the second argument takes Seconds, and the const being passed is in Milliseconds. 
Giving my a recovery time of 833 days.

**Review by Rofrtd — COMMENTED** [review]
Working fine, just need the review the recovery time.

File: src/features/game/events/landExpansion/drillOilReserve.ts (hunk 1 above)
Comment (spencerdezartsmith) [reply src/features/game/events/landExpansion/drillOilReserve.ts:23]: Nice catch. I think this snuck back in while playing around with time while testing 😱

**Pull request merged**